### ISSUE TITLE
An error message is displayed using the cis-QTL mapping: conditionally independent QTLs function

### ISSUE THREAD

**Issue by hyj260** [broadinstitute/tensorqtl#171]
When I ran the example files, everything worked fine. However, when I used my own data for cis-QTL mapping with permutations, I received numerous warnings, such as:

processing phenotype 3846/15661 WARNING: scipy.optimize.newton failed to converge (running scipy.optimize.minimize)
processing phenotype 3885/15661 WARNING: scipy.optimize.newton failed to converge (running scipy.optimize.minimize)
and
/public/home/hanzhijie/.local/lib/python3.8/site-packages/tensorqtl/core.py:315: RuntimeWarning: invalid value encountered in sqrt
  return 2*stats.t.cdf(-np.abs(np.sqrt(tstat2)), dof)

Despite these warnings, the cis-QTL mapping with permutations was able to generate an output file.
However, when I performed cis-QTL mapping for conditionally independent QTLs, the process failed to produce an output file and returned the following error:

<frozen importlib._bootstrap>:219: RuntimeWarning: scipy._lib.messagestream.MessageStream size changed, may indicate binary incompatibility. Expected 56 from C header, got 64 from PyObject
Mapping files: 100%|██████████| 3/3 [00:06<00:00,  2.00s/it]
/public/home/hanzhijie/.local/lib/python3.8/site-packages/tensorqtl/genotypeio.py:156: PerformanceWarning: Slicing is producing a large chunk. To accept the large
chunk and silence this warning, set the option
    >>> with dask.config.set(**{'array.slicing.split_large_chunks': False}):
    ...     array[indexer]
To avoid creating the large chunks, set the option
    >>> with dask.config.set(**{'array.slicing.split_large_chunks': True}):
    ...     array[indexer]
  self.bed = self.bed[:,ix]
Traceback (most recent call last):
  File "/public/software/apps/anaconda3/2021.05/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 3621, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 136, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 163, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 5198, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 5206, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'qval'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/public/software/apps/anaconda3/2021.05/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/public/software/apps/anaconda3/2021.05/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/public/home/hanzhijie/.local/lib/python3.8/site-packages/tensorqtl/__main__.py", line 2, in <module>
    tensorqtl.main()
  File "/public/home/hanzhijie/.local/lib/python3.8/site-packages/tensorqtl/tensorqtl.py", line 231, in main
    res_df = cis.map_independent(genotype_df, variant_df, summary_df, phenotype_df, phenotype_pos_df, covariates_df,
  File "/public/home/hanzhijie/.local/lib/python3.8/site-packages/tensorqtl/cis.py", line 797, in map_independent
    signif_df = cis_df[cis_df[fdr_col] <= fdr].copy()
  File "/public/software/apps/anaconda3/2021.05/lib/python3.8/site-packages/pandas/core/frame.py", line 3505, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/public/software/apps/anaconda3/2021.05/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 3623, in get_loc
    raise KeyError(key) from err
KeyError: 'qval'
[Dec 06 19:13:02] Running TensorQTL: cis-QTL mapping
  * WARNING: using CPU!
  * reading phenotypes (/public/home/hanzhijie/heyijie/paper2/single-cell-eqtl/data/expression/110/CRBL.Endo.bed)
  * cis-window detected as position ± 1,000,000
  * reading covariates (/public/home/hanzhijie/heyijie/paper2/single-cell-eqtl/data/cov/CRBL.Endo-cov.txt)
  * loading genotypes

I suspect that the issue might be due to the genotype file I created myself. I used the following code to convert the VCF file into a BED file:
/public/home/hanzhijie/software/plink1/plink --make-bed --output-chr chrM --vcf /public/home/hanzhijie/heyijie/paper2/single-cell-eqtl/data/genotype/final_file3.txt --keep-allele-order --out /public/home/hanzhijie/heyijie/paper2/single-cell-eqtl/data/genotype/out

My VCF file is formatted as follows:
##fileformat=VCFv4.2											
##FILTER=<ID=PASS,Description="All	filters	passed">									
##INFO=<ID=DP,Number=1,Type=Integer,Description="Read	Depth">										
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">											
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	X00-38	X01-37	X02-06
X	100000155	rs34240145	GT	G	.	.	.	GT	0/1	0/1	1/1
X	100000572	rs1116035	G	A,C	.	.	.	GT	0/1	0/1	1/1
X	100003723	rs1462585	G	A	.	.	.	GT	0/1	0/1	1/1
X	100004060	rs5920788	C	T	.	.	.	GT	0/1	0/1	1/1
X	100006662	rs5921452	T	C	.	.	.	GT	0/1	0/1	1/1

**Comment by francois-a** [comment]
Hi, the warnings can be ignored (when they occur a slightly slower optimization is used) but should largely disappear in v1.0.10. The input for the conditional mapping should be the output from the initial step with permutations, and needs to include the 'qval' column, which the error says is missing. Make sure you have R and rpy2 installed when running the permutations; you can also compute these separately using [`post.calculate_qvalues`](https://github.com/broadinstitute/tensorqtl/blob/master/tensorqtl/post.py#L17).

**Comment by hyj260** [comment]
Thank you for your help! The issue has been resolved. It turned out that when running the permutations, I had not properly installed rpy2 and overlooked the warning: "Warning: 'rfunc' cannot be imported. R with the 'qvalue' library, and the 'rpy2' Python package are needed." Since the permutations step generated an output file, I mistakenly assumed it had run correctly. However, the permutations output did not include the 'qval' and 'pval_nominal_threshold' columns. As a result, even after correctly installing 'qvalue' and 'rpy2', I encountered errors during the conditional mapping step.

I would like to offer a small suggestion: during the permutations step, if 'qvalue' or 'rpy2' are not properly installed, the program should throw an error and halt execution rather than proceed and generate incomplete results.

Finally, thank you for developing such excellent software and for your prompt and attentive responses to user inquiries.
